### PR TITLE
fix(gulp): prevent duplicate error messages

### LINKED ISSUE #2021: Remove duplicate error messages from the build
currently there are several layers in our build system that log and rethrow exceptions, we need to unify these and suppress errors (or stack traces for errors) so that the output is easy to understand and doesn't contain duplicate messages or useless stack traces.

Sources of exceptions/exception logging:
- broccoli plugins
- angular_builder.ts
- gulp's runSequence
- gulp's tasks

common scenarios to tackle:

interactive/watch mode:
- [x] typescript compilation error
- [x] traceur compilation error
- [x] karma browser test failure/error (`gulp test.unit.js`)
- [x] jasmine test failure/error (`gulp test.unit.cjs`)
- [ ] ts2dart errors
- [x] dart unit tests failures/errors

ci mode:
- [x] typescript compilation error
- [x] traceur compilation error
- [x] karma browser test failure/error (`gulp test.unit.js`)
- [x] jasmine test failure/error (`gulp test.unit.cjs`)
- [x] ts2dart errors
- [x] dart unit tests failures/errors

WIP branch resolving some errors: https://github.com/angular/angular/compare/master...IgorMinar:tripple-errors?expand=1

### PR DESCRIPTION
**Pull request by btford**
Closes #2021

### CODE REVIEW

**Comment by tbosch** [comment]
@IgorMinar could you review?

File: gulpfile.js [hunk 1]
```diff
@@ -203,11 +215,35 @@ gulp.task('build/pubbuild.dart', pubbuild(gulp, gulpPlugins, {
...
+  proc.stderr.pipe(through2(function (chunk, enc, callback) {
+    buffer += chunk.toString();
+    callback();
+  }, function (cb) {
+    var lines = buffer.split('\n');
+    var match, line;
+    var leadingWhitespace;
+    for (var i = 0; i < lines.length; i += 1) {
+      line = lines[i];
+      if (match = stripAnsi(line).match(ARROW_LINE)) {
```
Comment (IgorMinar) [inline gulpfile.js]: what's the purpose of stripAnsi? wouldn't this all the formatting from the dart formatter output?

Comment (btford) [inline gulpfile.js]: The regex looking for lines like `    ^^^^` uses color codes which otherwise cause the regex to fail.

**Comment by btford** [comment]
This should be good, but I need #2256 in first.

File: tools/broccoli/angular_builder.ts [hunk 2]
```diff
@@ -86,17 +86,6 @@ export class AngularBuilder {
...
-          console.error(err.toString());
-          // Should show file and line/col if present
-          if (err.file) {
-            console.error('File: ' + err.file);
-          }
-          if (err.stack) {
-            console.error(err.stack);
-          }
-          throw err;
-        });
```
Comment (IgorMinar) [inline tools/broccoli/angular_builder.ts]: don't we also need to mark the errors thrown from tsc and other plugins as `showStack=false`

**Comment by btford** [comment]
Landed as 31b6687894dbba26088c8af819436dcfb0c250e3 and 381d4cb30a73e16e270f0d1e90e6a6c7d6241512

**Pull request closed without merging**